Controllers/WishListsController.cs

```diff
@@ -19,44 +20,64 @@ namespace Nwazet.Commerce.Controllers {
     public class WishListsController : Controller {
         private readonly IWorkContextAccessor _wca;
         private readonly IWishListServices _wishListServices;
+        private readonly IWishListsUIServices _wishListsUIServices;
         private readonly IContentManager _contentManager;
         private readonly IEnumerable<IProductAttributeExtensionProvider> _attributeExtensionProviders;
         private readonly IShoppingCart _shoppingCart;
         private readonly IWorkflowManager _workflowManager;
         private readonly IEnumerable<IWishListExtensionProvider> _wishListExtensionProviders;
+        private readonly IOrchardServices _orchardServices;
+        private readonly IMembershipService _membershipService;
 
         public WishListsController(
             IWorkContextAccessor wca,
             IWishListServices wishListServices,
+            IWishListsUIServices wishListsUIServices,
             IContentManager contentManager,
             IEnumerable<IProductAttributeExtensionProvider> attributeExtensionProviders,
             IShoppingCart shoppingCart,
             IWorkflowManager workflowManager,
-            IEnumerable<IWishListExtensionProvider> wishListExtensionProviders) {
+            IEnumerable<IWishListExtensionProvider> wishListExtensionProviders,
+            IOrchardServices orchardServices,
+            IMembershipService membershipService) {
 
             _wca = wca;
             _wishListServices = wishListServices;
+            _wishListsUIServices = wishListsUIServices;
             _contentManager = contentManager;
             _attributeExtensionProviders = attributeExtensionProviders;
             _shoppingCart = shoppingCart;
             _workflowManager = workflowManager;
             _wishListExtensionProviders = wishListExtensionProviders;
+            _orchardServices = orchardServices;
+            _membershipService = membershipService;
+
+            T = NullLocalizer.Instance;
         }
 
+        public Localizer T { get; set; }
+
         private const string AttributePrefix = "productattributes.a";
         private const string ExtensionPrefix = "ext.";
 
         [OutputCache(Duration = 0)]
+        public ActionResult Index(string uName, int id = 0) {
```

Comment:
Shouldn't "Index" be a view of all your wish lists, and we'd have a separate action to view a specific list?